apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -129,6 +135,9 @@ def __init__(
         """The TileDB context to use, either provided or lazily constructed."""
         self._timestamp_ms = _maybe_timestamp_ms(timestamp)
 
+        """User specified threadpool. If None, we'll instantiate one ourselves."""
+        self._threadpool = threadpool or futures.ThreadPoolExecutor()
```

Comment:
We could change the implementation so, if the thread pool isn't specified by the user, then it is created lazily. Not sure it's worth it as the overhead should be minimal.

Reply:
Yeah, just creating a threadpool doesn’t actually start any threads until they are called for. This is 100% fine.